services/src/datasets/listing.rs

```diff
@@ -0,0 +1,120 @@
+use crate::datasets::storage::{AddDataSetProvider, AddMockDataSetProvider, DataSet};
+use crate::error::Result;
+use crate::users::user::UserId;
+use crate::util::user_input::{UserInput, Validated};
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::DataSetId;
+use geoengine_operators::engine::{
+    MetaData, MetaDataProvider, ResultDescriptor, VectorResultDescriptor,
+};
+use geoengine_operators::mock::MockDataSetDataSourceLoadingInfo;
+use geoengine_operators::source::OgrSourceDataset;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct DataSetListing {
+    pub id: DataSetId,
+    pub name: String,
+    pub description: String,
+    pub tags: Vec<String>,
+    pub source_operator: String, // TODO: enum?
```

Comment:
Hm, this is unclear, an enum would be bad for extensibility.
I guess a string is okay for now, maybe a type id?…

Reply:
so you mean https://doc.rust-lang.org/std/any/struct.TypeId.html ? it doesn't seem to be serializable